include/aspect/heating_model/interface.h

```diff
@@ -64,14 +64,36 @@ namespace aspect
                            const unsigned int n_comp);
 
       /**
-       * All source terms of the temperature equation at the given position.
-       * This includes shear heating, adiabatic heating, radiogenic heat
-       * production, the right hand side part of latent heat or any other
-       * heating terms on the right hand side of the energy equation.
+       * All source terms of the temperature equation that represent a
+       * continuous process leading to a rate of change in temperature
+       * at the given position. This includes shear heating, adiabatic
+       * heating, radiogenic heat production, or any other heating rates
+       * on the right hand side of the energy equation.
        */
       std::vector<double> heating_source_terms;
 
       /**
+       * The source terms of the temperature equation that represent
+       * fast changes in temperature (compared to the advection time
+       * scale), for example due to reactions, at the given position.
+       * On the advection time scale, these reactions might look like
+       * instantaneous changes in temperatures. This includes for example
+       * latent heat of melt.
+       *
+       * These reaction rates are only used in the operator_splitting nonlinear
+       * solver scheme, which allows it to solve reactions of compositional
+       * fields and temperature decoupled from the advection, and using a
+       * different time step size.
+       * In this case, they are used in addition to (and independent from) any
+       * heating_source_terms that a heating model defines, which are assembled
+       * as usual. For any other solver scheme, these values are ignored.
+       *
+       * In contrast to the heating source terms, these terms are actual changes
+       * in temperature (units K/s or K/yr) rather than changes in energy.
+       */
+      std::vector<double> rates_of_temperature_change;
```

Comment:
As mentioned in a previous comment, I'm not excited about having two fields with a similar meaning in this structure, in particular since they are not both used by all schemes. Can we find a way so that the material model doesn't have to know what solver is used?